### PR TITLE
Helper functions for Namespace labelling tests

### PR DESCRIPTION
**Pull request by vpinisetti-px**
**What this PR does / why we need it**:
These helper functions required to automate namespace label tests

**Which issue(s) this PR fixes** (optional)
Closes # https://portworx.atlassian.net/browse/PA-687

**Special notes for your reviewer**:

CreateBackupWithNamespaceLabel
http://aetos.pwx.purestorage.com/resultSet/testSetID/129888

CreateScheduleBackupWithNamespaceLabel
http://aetos.pwx.purestorage.com/resultSet/testSetID/129899

### CODE REVIEW

File: tests/backup/backup_helper.go
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1]: 1. CreateBackupWithNamespaceLabel and CreateScheduleBackupWithNamespaceLabel functions do not make use of backupSuccessCheck and restoreSuccessCheck helper functions. Please refer to PR #1215.

2. Could you make these helper functions without a success check, like CreateBackupWithoutCheck and CreateScheduleBackupWithoutCheck?

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -1526,3 +1526,154 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
...
+	}
+	_, err = backupDriver.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return err
+	}
+	log.Infof("Successfully created Backup [%s] with namespace label", backupName)
+	return nil
+}
+
+// CreateScheduleBackupWithNamespaceLabel creates a backup with namespace label
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1609]: creates a schedule* backup

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -1526,3 +1526,154 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
...
+		}
+		actual := resp.GetBackup().GetStatus().Status
+		expected := api.BackupInfo_StatusInfo_Success
+		if actual != expected {
+			return "", true, fmt.Errorf("status for first schedule backup [%s] expected was [%s] but got [%s]", firstScheduleBackupName, expected, actual)
+		}
+		return "", false, nil
+	}
+
+	_, err = task.DoRetryWithTimeout(backupSuccessCheck, 10*time.Minute, 30*time.Second)
```
Comment (snigdha-px) [inline tests/backup/backup_helper.go:1674]: Can we use the same variable : maxWaitPeriodForBackupCompletionInMinutes here as well?

File: tests/backup/backup_helper.go
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go:1]: Done @kphalgun-px

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go]: Done

File: tests/backup/backup_helper.go (hunk 2 above)
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go]: Done

File: tests/backup/backup_helper.go [hunk 3]
```diff
@@ -1445,3 +1445,194 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
...
+	}
+	_, err := backupDriver.CreateBackup(ctx, bkpCreateRequest)
+	if err != nil {
+		return err
+	}
+	err = backupSuccessCheck(backupName, orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
+	if err != nil {
+		return err
+	}
+	log.Infof("Successfully created backup [%s] with namespace label", backupName)
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1488]: How about we include label in the log?

File: tests/backup/backup_helper.go [hunk 4]
```diff
@@ -1445,3 +1445,194 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
...
+			Name: postRuleName,
+			Uid:  postRuleUid,
+		},
+		NsLabelSelectors: namespaceLabel,
+	}
+	_, err := backupDriver.CreateBackupSchedule(ctx, bkpSchCreateRequest)
+	if err != nil {
+		return err
+	}
+	time.Sleep(1 * time.Minute)
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1529]: I think I forgot to remove the wait in CreateScheduleBackup as well. Could you please remove this unnecessary wait in both the functions?

File: tests/backup/backup_helper.go [hunk 5]
```diff
@@ -1445,3 +1445,194 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
...
+	firstScheduleBackupName, err = GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+	if err != nil {
+		return err
+	}
+	err = backupSuccessCheck(firstScheduleBackupName, orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
+	if err != nil {
+		return err
+	}
+	log.Infof("Successfully created schedule backup [%s] with namespace label", firstScheduleBackupName)
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:1538]: How about we include label in the log?

File: tests/backup/backup_helper.go (hunk 3 above)
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go:1488]: We can add.. But when you add multiple labels(eg: 100) the string would be too big. Will discuss with the team and take a decision.

File: tests/backup/backup_helper.go (hunk 5 above)
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go:1537]: done

File: tests/backup/backup_helper.go (hunk 4 above)
Comment (vpinisetti-px) [reply tests/backup/backup_helper.go]: done

Comment (kphalgun-px) [reply tests/backup/backup_helper.go]: Could you please remove it in the original CreateScheduleBackup as well?

**Review by snigdha-px — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]
lgtm

**Review by kshithijiyer-px — APPROVED** [review]
LGTM! Resolve the merge conflict and merge the patch.

File: tests/backup/backup_helper.go (hunk 3 above)
Comment (kphalgun-px) [reply tests/backup/backup_helper.go]: Thanks!

File: tests/backup/backup_helper.go (hunk 4 above)
Comment (kphalgun-px) [reply tests/backup/backup_helper.go]: I will take care of this in my next PR. Thanks!

File: tests/backup/backup_helper.go (hunk 5 above)
Comment (kphalgun-px) [reply tests/backup/backup_helper.go]: Thanks!

**Pull request merged**